rubygems/index.js

```diff
@@ -0,0 +1,230 @@
+    const name = pkgParts[0].replace('gem', '').trim().replace(/'/g, '').replace(/"/g, '')
+
+    // If there's nothing after the name, then return latest as version
+    if (!pkgParts[1]) {
+      return {
+        name,
+        operator: '==',
+        versionSpec: 'latest',
+        toString: () => `${name}@`
+      }
+    }
+
+    const re = /^('|")(==|=|>=|>|<=|~>)?\s*([0-9.]+)('|")$/
+    const match = pkgParts[1].trim().match(re)
+    if (!match) throw new Error(`unparseable pkg dependency ${pkgParts[1]}`)
+    const operator = match[2] || '='
+    const version = match[3]
+
+    return {
+      name,
+      operator,
+      versionSpec: version,
+      toString: () => `${name}@${operator}${version}`
+    }
+  }
+
+  // To list versions - hit https://rubygems.org/api/v1/versions/[gem name].json
+  // To get deps of specific version - hit https://rubygems.org/api/v2/rubygems/[gem name]/versions/[version].json
+  // to get latest version deps - hit https://rubygems.org/api/v2/rubygems/[gem name].json
+  async getDependencies(pkgSpec) {
+    let name
+    let version
+
+    try {
+      const resolved = await this.resolve(pkgSpec)
+      name = resolved.name
+      version = resolved.version
+    } catch (e) {
+      this.log.error(e)
+      // unable to resolve the given spec; no way to get the deps for this input
+      return []
+    }
+
+    let body
+    if (!version) {
+      ({ body } = await this.got(`https://rubygems.org/api/v2/rubygems/${name}.json`, {
+        responseType: "json"
+      }))
+    } else {
+      ({ body } = await this.got(
+        `https://rubygems.org/api/v2/rubygems/${name}/versions/${version}.json`,
+        {
+          responseType: "json"
+        }
+      ))
+    }
+
+    const dependencyKeys = Object.keys(body.dependencies)
+    const depRequirements = dependencyKeys.reduce((acc, key) => {
+      const runtime = body.dependencies[key]
+      return acc.concat(runtime.reduce((ac, dep) => {
+        // each dep is formatted like this
+        /**
+         * {
+              "name": "activerecord",
+              "requirements": "= 3.0.18"
+          }
+        */
+        const versionSplit = dep.requirements.split(' ')
+        const spec = {
+          name: dep.name,
+          version: versionSplit[1],
+          operator: versionSplit[0],
+          toString: () => `${dep.name}@${versionSplit[0]}${versionSplit[1]}`
+        }
+        return ac.concat(spec)
+      }, []))
+    }, [])
+
+    if (!depRequirements || !depRequirements.length) return []
+
+    return depRequirements
+  }
+
+  // input: output of getSpec
+  // resolves the most suitable version to freeze given <name><operator><version>
+  // for example: rubocop >= 3.0.0 would fetch all versions available, and select the highest version
+  async resolve(pkgSpec) {
+    const { name, operator, versionSpec: version } = pkgSpec
+
+    if (operator === "==" || operator === '=') {
```

Comment:
is this a conditional imposed upon ourselves by the getSpec func? or are `==` and `=` both valid operators in a Gemfile / API response ?

Reply:
i'm second guessing myself but in response i see '= 1.0.0' and in gemfile i'm pretty sure i saw '== 1.0.0'